### DISCUSSION TITLE
Set resolution using api or set default resolution

### DISCUSSION THREAD

**Discussion by sidelux** [jhomlala/betterplayer#721] [Ideas]
Hi,
i would like to know if there is a way to set a custom resolution before building the player (not the first one only), or an api to set the resolution in a later moment (something like `setResolution("resolution name")`).
At the end i need to "save" the resolution setting for all my videos without reselect every time.
Thanks

**Comment by jhomlala** [comment]
Continuation of https://github.com/jhomlala/betterplayer/issues/717
What's your data source? Is it HLS or non-HLS?

**Comment by sidelux** [comment]
It's an MP4 file so a non-HLS, i have 3 resolution videos (480p, 720p, 1080p).

**Reply by jhomlala** [reply]
So why this doesn't work for you? https://jhomlala.github.io/betterplayer/#/resolutionsofvideo

**Reply by sidelux** [reply]
It's work the resolution selector, but i need to change the default resolution (the one where the players open) based on user preference.
So if i have the "480p" as default, if the user set "720p" for a video, the next video will have the "720p" resolution by default and he doesn't need to change every video.
Basically i need to "save" the resolution setting for all videos instead to change it every time, but for a user could be the 480, for another could be 1080., etc...

**Reply by jhomlala** [reply]
You should save user chosen option in shared prefs and then read it when needed. To change resolution from your code, please use `setResolution` method from `BetterPlayerController`. 

In the future I will add url of the picked resolution in the `BetterPlayerEventType.changedResolution` event.

**Reply by sidelux** [reply]
I need to use my resolution name as parameter? I can't find a documentation about it.
Something like setResolution("720p").

**Reply by jhomlala** [reply]
You need to pass url of the video in the `setResolution` method.

**Reply by sidelux** [reply]
Ok so i need to wait the url in event to save the resolution.
Thanks
